lib/multitenancy_tools/dump_cleaner.rb

```diff
@@ -17,7 +18,23 @@ def clean
       @sql.gsub!(/SET lock_timeout .*;\n/, '')
       @sql.gsub!(/^--(?:.*)\n+/, '')
       @sql.gsub!(/\n+/, "\n")
+      clean_schema_names!
       @sql
     end
+
+    private
+
+    # Matches namespaces containing the given schema_name
+    #
+    # For instance:
+    # CREATE TABLE schema1.posts ();
+    #
+    # Will be replaced by:
+    # CREATE TABLE posts ();
+    def clean_schema_names!
+      return if @schema_name.blank?
+
+      @sql.gsub!(/#{@schema_name}\.(\w*)/, '\1')
```

Comment:
Can we at least add a word delimiter before the schema name? Is it escaped when present in the file like with a quote or something?

Reply:
I made some tests and it is not quoting schema names, 'cause we don't pass the `--quote-all-identifiers` option into it.

I'm not sure if we can add a delimiter to `schema` name at all, because it's generation is completely handle by `pg_dump` itself. The only option I see is to use different schema names in the clients, but that is not suitable for our purposes... Any ideas on this?

Reply:
I've added a little more scenarios in the unit specs to grant more reliability on this. Also made some changes in the regex itself to cover some flaws in the previous matching pattern.